docker-k3os-image-gen/docker-compose.yml

```diff
@@ -0,0 +1,33 @@
+version: "3"
+services:
+  rpi-image-gen:
+    build: ../
+    environment:
+      TARGET: raspberrypi
+    volumes:
+      # bind ../config to our working dir config
+      # this ONLY works with unnamed volumes like below!!
+      - ../config:/usr/src/app/config
+      # result is in this named volume
+      - "images:/usr/src/app/images"
+      # this is the downloads cache
+      - "k3os-fw:/usr/src/app/deps"
+      # this is needed for loopback devices
+      - /dev:/dev
+    privileged: true
+    devices:
+      - "/dev:/dev"
+    cap_add:
+      - CAP_MKNOD
+volumes:
+   k3os-fw:
+     driver: local
+   images:
+     # OSX only!! if we add this the below path under device is where the volume will reside actually on OSX
+     # otherwise it is in the docker for OSX vm, where we have ONLY indirect access via other containers or command line ...
```

Comment:
This would, in principle, be supported for any OS, right? Not just Docker for Mac? E.g. Docker for Windows, and just native Linux Docker, would also allow building this way? In this case, perhaps we could remove the references to OSX and just mention in the README that this is a possible way to build the image on Windows, Mac or Linux without needing local dependencies?

Reply:
named volumes with bind mount work perfectly under OSX, somehow under Windows and are not needed for Linux ... :-)
Maybe this changed in the last months, since there is now a new option [mutagen chaching](https://docs.docker.com/docker-for-mac/mutagen-caching/ ) but testing this on all 3 platforms is a lot of work...